docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -63,7 +71,91 @@ def downstream_asset(upstream_asset):
     return upstream_asset + [4]
 ```
 
-The [explicit dependencies](#explicit-dependencies) example covers an alternative way to specify asset dependencies without needing to match argument names to upstream asset names.
+#### Defining explicit dependencies
+
+If defining dependencies by matching argument names to upstream asset names feels too magical for your tastes, you can also define dependencies in a more explicit way:
+
+```python file=/concepts/assets/explicit_asset_dependency.py
+from dagster import AssetIn, asset
+
+
+@asset
+def upstream_asset():
+    return [1, 2, 3]
+
+
+@asset(ins={"upstream": AssetIn("upstream_asset")})
+def downstream_asset(upstream):
+    return upstream + [4]
+```
+
+In this case, `ins={"upstream": AssetIn("upstream_asset")}` declares that the contents of the asset with the key `upstream_asset` will be provided to the function argument named `upstream`.
+
+#### Defining external asset dependencies
+
+Software-defined assets frequently depend on assets that are generated elsewhere. Using <PyObject object="SourceAsset" />, you can include these external assets and allow the assets in your <PyObject object="AssetGroup" /> to depend on them.
+
+For example:
+
+```python file=/concepts/assets/source_asset.py startafter=start_marker endbefore=end_marker
+from dagster import AssetGroup, AssetKey, SourceAsset, asset
+
+my_source_asset = SourceAsset(key=AssetKey("my_source_asset"))
+
+
+@asset
+def my_derived_asset(my_source_asset):
+    return my_source_asset + [4]
+
+
+asset_group = AssetGroup(assets=[my_derived_asset], source_assets=[my_source_asset])
+```
+
+You can also re-use assets across repositories by including them as source assets:
+
+```python file=/concepts/assets/cross_repository_asset.py
+from dagster import AssetGroup, AssetKey, Output, SourceAsset, asset, repository
+
+
+@asset
+def my_source_asset():
+    yield Output(5)
+
+
+@asset
+def second_asset(my_source_asset):
+    yield Output(1)
+
+
+upstream_asset_group = AssetGroup(assets=[my_source_asset, second_asset])
+
+
+@repository
+def upstream_repo():
+    return [upstream_asset_group]
+
+
+my_source_asset = SourceAsset(key=AssetKey("my_source_asset"))
+
+
+@asset
+def downstream_asset(my_source_asset):
+    yield Output(5)
+
+
+asset_group = AssetGroup(assets=[downstream_asset], source_assets=[my_source_asset])
+
+
+@repository
+def downstream_repo():
+    return [asset_group]
+```
+
+Using source assets has a few advantages over having the code inside of an asset's op load the data:
+
+- **Dagit can show asset lineage that includes the source assets**. If a different <PyObject object="AssetGroup" /> in the same [workspace](/concepts/repositories-workspaces/workspaces) contains an asset definition with the same asset key as a <PyObject object="SourceAsset" />, Dagit can represent the asset lineage across those groups.
+- **Dagster can use data-loading code facotred into an <PyObject object="IOManager" /> to load the contents of the source asset**.
```

Comment:
Nit: facotred -> factored

Reply:
Spelling errors are never nitpicks! Thanks for catching this.